clang/lib/Driver/Driver.cpp

```diff
@@ -5970,7 +5970,10 @@ bool clang::driver::isObjectFile(std::string FileName) {
   if (!llvm::sys::path::has_extension(FileName))
     // Any file with no extension should be considered an Object. Take into
     // account -lsomelib library filenames.
-    return FileName.rfind("-l", 0) != 0;
+    // When -lstdc++ is passed in certain cases it's transformed into an driver 
+    // option.   
+    return (FileName.rfind("-l", 0) != 0 && 
+            FileName != "-Z-reserved-lib-stdc++"); 
```

Comment:
I lack knowledge if `-Z` transformations of library names can be a recurring reserved case (your comment in the PR does highlight one more example, though).

There are two choices:
1. Leave the check against the full name and assume that we're unlikely to run into any similar case
2. Replace with `FileName.rfind("-Z", 0) != 0`, thereby banishing the user from having a regular filename start this way.

`-l` specifics are standard across compilers, but is 2 violating the industry standards? Can it be done w/o documenting in user manual? @mdtoguchi @Fznamznon @AlexeySachkov can someone please enlighten me here?

Reply:
I think the easiest is option 1, there appears to be only 2 reserved libraries for now (so it seems uncommon at least): 
https://github.com/intel/llvm/blob/sycl/clang/include/clang/Driver/Options.td#L3193 

Using a similar method to the libraries by rfind'ng on "-Z" would likely run the risk of running into some issues as it appears that -Z is one way to pass arguments to the linker through the driver (from what I can gather from https://clang.llvm.org/docs/ClangCommandLineReference.html) and there is at least one other -Z option that's unrelated to a reserved library (a few lines above the source link I gave).

As a maybe useful aside the transformation of the library arguments to options occurs here:
https://github.com/intel/llvm/blob/sycl/clang/lib/Driver/Driver.cpp#L377